Demo/Demo/ViewModels/PayPalWebViewModel.swift

```diff
@@ -82,6 +82,45 @@ class PayPalWebViewModel: ObservableObject, PayPalWebCheckoutDelegate {
         }
     }
 
+    func captureOrder(orderID: String, selectedMerchantIntegration: MerchantIntegration) async throws {
+        do {
+            DispatchQueue.main.async {
+                self.state.capturedOrderResponse = .loading
+            }
+            let order = try await DemoMerchantAPI.sharedService.captureOrder(
+                orderID: orderID,
+                selectedMerchantIntegration: selectedMerchantIntegration
+            )
+            DispatchQueue.main.async {
+                self.state.capturedOrderResponse = .loaded(order)
+            }
+        } catch {
+            DispatchQueue.main.async {
+                self.state.capturedOrderResponse = .error(message: error.localizedDescription)
+            }
+            print("Error capturing order: \(error.localizedDescription)")
+        }
+    }
+
+    func authorizeOrder(orderID: String, selectedMerchantIntegration: MerchantIntegration) async throws {
+        do {
+            DispatchQueue.main.async {
+                self.state.authorizedOrderResponse = .loading
+            }
+            let order = try await DemoMerchantAPI.sharedService.authorizeOrder(
+                orderID: orderID,
+                selectedMerchantIntegration: selectedMerchantIntegration
+            )
+            DispatchQueue.main.async {
+                self.state.authorizedOrderResponse = .loaded(order)
+            }
+        } catch {
+            DispatchQueue.main.async {
+                self.state.authorizedOrderResponse = .error(message: error.localizedDescription)
+            }
+            print("Error authorizing order: \(error.localizedDescription)")
+        }
+    }
```

Comment:
Same here, this looks like a copy-paste from `CardPaymentViewModel` ([see snippet](https://github.com/paypal/paypal-ios/blob/a490ffdc891f5b5c3137749655f08bd9c7513339/Demo/Demo/ViewModels/CardPaymentViewModel.swift#L61-L99)). That duplication is usually a good sign that we should refactor.

Happy to talk through some refactor options. We could go the route of a parent class or a protocol to represent a generic `PaymentMethodViewModel` that has a set of default functionality that is re-usable across all payment methods (ex: auth, capture).

Reply:
I'll make protocols from the smallest element up (from ActionButton to PaymentMethodViewModel) and break them up into different commits.

Reply:
@scannillo , I looked into making a PaymentMethodViewModel protocol extension to provide common authorizeOrder and captureOrder functions and I ran into issue of having to change stored properties in the conforming types. Protocol extensions, as I understand, cannot mutate stored properties of conforming types. 

I ran into similar issue with factoring out views that looks the same -` state` in each ViewModel is of different type. I would need to use associated types and also run into issue of observing different properties of states of different types for each viewModels. 

I think complexity introduced to save some duplication outweighs the benefits here.

Reply:
I will continue working with the wrapper protocol for `LoadingState`. I guess all both flows have views for order creation result, order approval result and order capture/authorize result. I can pass in payment method to toggle the values.

Reply:
I created another ticket for refactoring common views and functionalities in the viewModels. https://paypal.atlassian.net/browse/DTPPCPSDK-1437